velox/vector/FlatVector-inl.h

```diff
@@ -158,35 +164,56 @@ void FlatVector<T>::copyValuesAndNulls(
   }
 
   if (source->isFlatEncoding()) {
-    auto* sourceValues = source->typeKind() != TypeKind::UNKNOWN
-        ? source->asUnchecked<FlatVector<T>>()->rawValues()
-        : nullptr;
-    if (toSourceRow) {
-      rows.applyToSelected([&](auto row) {
-        auto sourceRow = toSourceRow[row];
-        if (sourceValues) {
-          rawValues_[row] = sourceValues[sourceRow];
-        }
-        if (rawNulls) {
-          bits::setNull(
-              rawNulls,
-              row,
-              sourceNulls && bits::isBitNull(sourceNulls, sourceRow));
-        }
-      });
+    auto* flatSource = source->asUnchecked<FlatVector<T>>();
+    if (flatSource->values() == nullptr) {
```

Comment:
nit: we can do this as a follow up, but wondering if it would be beneficial if we do all null copying at the beginning then only copy values for non-null rows